impl/src/planning/subplanners.rs

```diff
@@ -446,182 +441,163 @@ impl<'planner> CrateSubplanner<'planner> {
+      };
 
-  /** Yields the list of dependencies as described by the manifest (without version). */
-  fn identify_named_deps(&self) -> Result<(DependencyNames, HashMap<String, DependencyNames>)> {
-    // Resolve dependencies into types
-    let mut default_dep_names = DependencyNames {
-      build_dep_names: Vec::new(),
-      dev_dep_names: Vec::new(),
-      normal_dep_names: Vec::new(),
-      aliased_dep_names: HashMap::new(),
-    };
+      if !dep_set.aliased_dependencies.insert(dep_alias) {
+        return Err(
+          RazeError::Planning {
+            dependency_name_opt: Some(pkg.name.to_string()),
+            message: format!("Duplicated renamed package {}", name),
+          }
+          .into(),
+        );
+      }
+    }
 
-    let mut targeted_dep_names: HashMap<String, DependencyNames> = HashMap::new();
+    Ok(())
+  }
 
-    let package = self.crate_catalog_entry.package();
-    for dep in &package.dependencies {
-      // This shadow allow for dependencies with target restrictions to override where
-      // to write data about itself.
-      let mut dep_names = &mut default_dep_names;
+  fn buildable_target_for_dep(&self, dep_package: &Package) -> Result<String> {
+    // UNWRAP: Guaranteed to exist by checks in WorkspaceSubplanner#produce_build_plan
+    self
+      .crate_catalog
+      .entry_for_package_id(&dep_package.id)
+      .unwrap()
+      .workspace_path_and_default_target(&self.settings)
+  }
 
-      if dep.target.is_some() {
-        // UNWRAP: Safe from above check
-        let target_str = format!("{}", dep.target.as_ref().unwrap());
+  /// Test if the given dep details pertain to it being a proc-macro
+  ///
+  /// Implicitly dependencies are on the [lib] target from Cargo.toml (of which there is guaranteed
+  /// to be at most one).
+  ///
+  /// We don't explicitly narrow to be considering only the [lib] Target - we rely on the fact that
+  /// only one [lib] is allowed in a Package, and so treat the Package synonymously with the [lib]
+  /// Target therein. Only the [lib] target is allowed to be labelled as a proc-macro, so checking
+  /// if "any" target is a proc-macro is equivalent to checking if the [lib] target is a proc-macro
+  /// (and accordingly, whether we need to treat this dep like a proc-macro).
+  fn is_proc_macro(&self, dep_package: &Package) -> bool {
+    dep_package
+      .targets
+      .iter()
+      .flat_map(|target| target.crate_types.iter())
+      .any(|crate_type| crate_type.as_str() == "proc-macro")
+  }
 
-        // Legacy behavior
-        if let Some(platform_details) = &self.platform_details {
-          if let Some(settings_target) = &self.settings.target {
-            let platform = Platform::from_str(&target_str)?;
+  /// Test to see if a dep has been renamed
+  ///
+  /// Currently cargo-metadata provides rename detection in a few places, we take the names
+  /// from the resolution of a package
+  fn is_renamed(&self, dep_package: &Package) -> bool {
```

Comment:
Does it make sense to precompute some of these new properties? I'm lost in the details here, but part of the idea with the crate catalog was to make this type of lookup more efficient.

Reply:
Humm yeah might work, and might be a good way to make the find the root sillyness go away.